src/components/Tip.js

```diff
@@ -1,82 +1,94 @@
 import React, { useState } from 'react';
 import { StyleSheet, Text, View, Button, TextInput } from 'react-native';
 
-export default function Tip({amount}) {
-    const [tipPercentage, setTipPercentage] = useState(0)
-    const [useCustomTip, setUseCustomTip] = useState(false)
+export default function Tip({
+  amount,
+  setTipAmount
+}) {
+  const [tipPercentage, setTipPercentage] = useState(0)
+  const [useCustomTip, setUseCustomTip] = useState(false)
 
-    function isTipValid(amount) {
-        return String(amount).match(/^[0-9]+$/) && amount >= 0 && amount < 100
-    }
+  const isTipValid = (amount) => {
+    const numAmount = parseInt(amount);
+    return (numAmount >= 0) && (numAmount < 100);
+  }
 
-    function changePercentage(amount, isCustomPercentage) {
-        setTipPercentage(amount)
-        !isCustomPercentage && setUseCustomTip(false)
-    }
+  const changePercentage = (amount, tipPercentage, isCustomPercentage) => {
+    const tipAmount = (amount*tipPercentage/100).toFixed(2);
+    setTipPercentage(amount)
+    setTipAmount(tipAmount)
+    !isCustomPercentage && setUseCustomTip(false)
+  }
 
-    const validTipPercentages = [
-        {amt: 15, text: "15%"},
-        {amt: 18, text: "18%"},
-        {amt: 20, text: "20%"}
-    ]
+  const validTipPercentages = {
+    15: "15%",
+    18: "18%",
+    20: "20%"
+  }
```

Comment:
nit: hm, would there be a time where a tip percentage is not mapped to its string representation? if not, I think this can be an array of numbers and then map each component with a title of the string representation

Reply:
Hm.. yea I think I had a mapping to reduce the amount of work I would have to do to create a string representation since it's currently only being passed into the title prop of the Button component:
```JavaScript
<Button key={index}
   class='tip-percentage'
   title={validTipPercentageString}
   onPress={() => {
      if (isTipValid(validTipPercentageInt)) {
            changePercentage(amount, validTipPercentageInt, false)
      }
    }}
/>
```

I guess I can achieve the same by just passing the number, and have a method that formats the number into a percentage? Like this:
```JavaScript
const getValidTipPercentageString = (validTipPercentageInt) => {
   return `${validTipPercentageInt}%`
}

<Button key={index}
   class='tip-percentage'
   title={getValidTipPercentageString(validTipPercentageInt)}
   onPress={() => {
      if (isTipValid(validTipPercentageInt)) {
            changePercentage(amount, validTipPercentageInt, false)
      }
    }}
/>
```

WDYT? @cinderlee

Reply:
yup! that works for me